evidence.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2023 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package parsectpm
+
+import (
+	"crypto"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	cbor "github.com/fxamacker/cbor/v2"
+	"github.com/veraison/go-cose"
+)
+
+type Collection struct {
+	Kat *KAT `cbor:"kat" json:"kat"`
+	Pat *PAT `cbor:"pat" json:"pat"`
+}
+
+// Evidence is a wrapper around TpmParsec Attestation Collection
+type Evidence struct {
+	collection *Collection
+}
+
+func (e *Evidence) SetTokens(k *KAT, p *PAT) error {
+
+	if k == nil || p == nil {
+		return fmt.Errorf("nil token supplied")
+	}
+
+	if err := k.Validate(); err != nil {
+		return fmt.Errorf("validation of key attestation token failed: %w", err)
+	}
+
+	if err := p.Validate(); err != nil {
+		return fmt.Errorf("validation of platform attestation token failed: %w", err)
+	}
```

Comment:
This code seems to be reused and it's a good candidate for refactoring.

Maybe make it into a `check()` function on the Evidence object that you can call from different places.